zendesk/search.go

```diff
@@ -0,0 +1,127 @@
+package zendesk
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+// SearchOptions are the options that can be provided to the search API
+//
+// ref: https://developer.zendesk.com/rest_api/docs/support/search#available-parameters
+type SearchOptions struct {
+	PageOptions
+	Query     string `url:"query"`
+	SortBy    string `url:"sort_by,omitempty"`
+	SortOrder string `url:"sort_order,omitempty"`
+}
+
+type SearchAPI interface {
+	Search(ctx context.Context, opts *SearchOptions) (SearchResults, Page, error)
+}
+
+type SearchResults struct {
+	results []interface{}
+}
+
+func (r *SearchResults) MarshalJSON() ([]byte, error)  {
+	return json.Marshal(r.results)
+}
+
+func (r *SearchResults) UnmarshalJSON(b []byte) error  {
+	var results []interface{}
+	var tmp []json.RawMessage
+	err := json.Unmarshal(b, &tmp)
+	if err != nil {
+		return err
+	}
+
+	for _, v := range tmp {
+		// TODO: check with @nukosuke to see if we should try to speed this up using github.com/tidwall/gjson
```

Comment:
I was able to make this logic a decent amount faster by using the package mentioned above.

Here attached are my benchmark results. Let me know if you want me to add that logic slash bring in that package.

[benchmark.txt](https://github.com/nukosuke/go-zendesk/files/3938228/benchmark.txt)

Reply:
Thank you for the suggestion.
It's great result. But, I think it better to keep simple dependencies until the speed of JSON parsing would become bottleneck because this package is just an API library.

Reply:
At this time, we might want to add link to this conversation as `NOTE`.